functions/APIs/deals.js

```diff
@@ -0,0 +1,159 @@
+const { db } = require('../util/admin');
+
+/*
+display a list of all deals
+GET /deals
+success response: array of deal objects
+*/
+exports.getAllDeals = (request, response) => {
+    db.collection('deals').get()
+    .then((data) => {
+        let deals = [];
+        data.forEach((doc) => {
+            deals.push({
+                dealId: doc.id,
+                farmId: doc.data().farmId,
+                foodbankId: doc.data().foodbankId,
+                surplusId: doc.data().surplusId,
+                farmContactKey: doc.data().farmContactKey,
+                foodbankContactKey: doc.data().foodbankContactKey,
+                farmlinkContactName: doc.data().farmlinkContactName,
+                farmlinkContactPhone: doc.data().farmlinkContactPhone,
+                farmlinkContactEmail: doc.data().farmlinkContactEmail,
+                pickupDate: doc.data().pickupDate,
+                pickupTime: doc.data().pickupTime,
+                deliveryDate: doc.data().deliveryDate,
+                deliveryTime: doc.data().deliveryTime,
+                bill: doc.data().bill,
+                fundsPaidToFarm: doc.data().fundsPaidToFarm,
+                fundsPaidForShipping: doc.data().fundsPaidForShipping,
+                totalSpent: doc.data().totalSpent,
+                invoice: doc.data().invoice,
+                associatedPress: doc.data().associatedPress,
+                notes: doc.data().notes,
+            });
+        });
+        return response.json(deals);
+    })
+    .catch((err) => {
+        console.log(err);
+        return response.status(500).json({error: err.code});
+    });
+}
+
+/*
+display a specific deal
```

Comment:
Hm, I don't think these really display anything.  They retrieve the information, which you might then display, or do something else with.